<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5 Polygenic Adaptation in SLiM | Polygenic SLiMulations: Investigating polygenic adaptation with SLiM 3.</title>
  <meta name="description" content="5 Polygenic Adaptation in SLiM | Polygenic SLiMulations: Investigating polygenic adaptation with SLiM 3." />
  <meta name="generator" content="bookdown 0.22 and GitBook 2.6.7" />

  <meta property="og:title" content="5 Polygenic Adaptation in SLiM | Polygenic SLiMulations: Investigating polygenic adaptation with SLiM 3." />
  <meta property="og:type" content="book" />
  
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5 Polygenic Adaptation in SLiM | Polygenic SLiMulations: Investigating polygenic adaptation with SLiM 3." />
  
  
  

<meta name="author" content="Nick O’Brien" />


<meta name="date" content="2021-06-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="a-brief-overview-of-polygenic-adaptation.html"/>
<link rel="next" href="running-slim-in-parallel.html"/>
<script src="libs/header-attrs-2.8/header-attrs.js"></script>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="styles.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="introduction.html"><a href="introduction.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="introduction.html"><a href="introduction.html#preface"><i class="fa fa-check"></i><b>1.1</b> Preface</a></li>
<li class="chapter" data-level="1.2" data-path="introduction.html"><a href="introduction.html#overview"><i class="fa fa-check"></i><b>1.2</b> Overview</a></li>
<li class="chapter" data-level="1.3" data-path="introduction.html"><a href="introduction.html#prerequisites"><i class="fa fa-check"></i><b>1.3</b> Prerequisites</a></li>
<li class="chapter" data-level="1.4" data-path="introduction.html"><a href="introduction.html#references"><i class="fa fa-check"></i><b>1.4</b> References</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="installing-a-linux-environment-slim-on-the-wsl.html"><a href="installing-a-linux-environment-slim-on-the-wsl.html"><i class="fa fa-check"></i><b>2</b> Installing a Linux environment &amp; SLiM on the WSL</a>
<ul>
<li class="chapter" data-level="2.1" data-path="installing-a-linux-environment-slim-on-the-wsl.html"><a href="installing-a-linux-environment-slim-on-the-wsl.html#overview-1"><i class="fa fa-check"></i><b>2.1</b> Overview</a></li>
<li class="chapter" data-level="2.2" data-path="installing-a-linux-environment-slim-on-the-wsl.html"><a href="installing-a-linux-environment-slim-on-the-wsl.html#why-linux"><i class="fa fa-check"></i><b>2.2</b> Why Linux?</a></li>
<li class="chapter" data-level="2.3" data-path="installing-a-linux-environment-slim-on-the-wsl.html"><a href="installing-a-linux-environment-slim-on-the-wsl.html#the-windows-subsystem-for-linux"><i class="fa fa-check"></i><b>2.3</b> The Windows Subsystem for Linux</a></li>
<li class="chapter" data-level="2.4" data-path="installing-a-linux-environment-slim-on-the-wsl.html"><a href="installing-a-linux-environment-slim-on-the-wsl.html#installing-wsl"><i class="fa fa-check"></i><b>2.4</b> Installing WSL</a></li>
<li class="chapter" data-level="2.5" data-path="installing-a-linux-environment-slim-on-the-wsl.html"><a href="installing-a-linux-environment-slim-on-the-wsl.html#installing-a-desktop-and-setting-up-x11"><i class="fa fa-check"></i><b>2.5</b> Installing a desktop and setting up X11</a></li>
<li class="chapter" data-level="2.6" data-path="installing-a-linux-environment-slim-on-the-wsl.html"><a href="installing-a-linux-environment-slim-on-the-wsl.html#building-slim"><i class="fa fa-check"></i><b>2.6</b> Building SLiM</a></li>
<li class="chapter" data-level="2.7" data-path="installing-a-linux-environment-slim-on-the-wsl.html"><a href="installing-a-linux-environment-slim-on-the-wsl.html#installing-other-useful-apps"><i class="fa fa-check"></i><b>2.7</b> Installing other useful apps</a></li>
<li class="chapter" data-level="2.8" data-path="installing-a-linux-environment-slim-on-the-wsl.html"><a href="installing-a-linux-environment-slim-on-the-wsl.html#footnotes"><i class="fa fa-check"></i><b>2.8</b> Footnotes</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="terminal-shortcuts-and-basics.html"><a href="terminal-shortcuts-and-basics.html"><i class="fa fa-check"></i><b>3</b> Terminal Shortcuts and Basics</a>
<ul>
<li class="chapter" data-level="3.1" data-path="terminal-shortcuts-and-basics.html"><a href="terminal-shortcuts-and-basics.html#overview-2"><i class="fa fa-check"></i><b>3.1</b> Overview</a></li>
<li class="chapter" data-level="3.2" data-path="terminal-shortcuts-and-basics.html"><a href="terminal-shortcuts-and-basics.html#navigating-folders"><i class="fa fa-check"></i><b>3.2</b> Navigating folders</a></li>
<li class="chapter" data-level="3.3" data-path="terminal-shortcuts-and-basics.html"><a href="terminal-shortcuts-and-basics.html#installing-and-updating-software"><i class="fa fa-check"></i><b>3.3</b> Installing and updating software</a></li>
<li class="chapter" data-level="3.4" data-path="terminal-shortcuts-and-basics.html"><a href="terminal-shortcuts-and-basics.html#input-and-output"><i class="fa fa-check"></i><b>3.4</b> Input and output</a></li>
<li class="chapter" data-level="3.5" data-path="terminal-shortcuts-and-basics.html"><a href="terminal-shortcuts-and-basics.html#git"><i class="fa fa-check"></i><b>3.5</b> Git</a>
<ul>
<li class="chapter" data-level="3.5.1" data-path="terminal-shortcuts-and-basics.html"><a href="terminal-shortcuts-and-basics.html#git-branches-and-pull-requests"><i class="fa fa-check"></i><b>3.5.1</b> Git Branches and Pull Requests</a></li>
<li class="chapter" data-level="3.5.2" data-path="terminal-shortcuts-and-basics.html"><a href="terminal-shortcuts-and-basics.html#git-for-slim"><i class="fa fa-check"></i><b>3.5.2</b> Git for SLiM</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="a-brief-overview-of-polygenic-adaptation.html"><a href="a-brief-overview-of-polygenic-adaptation.html"><i class="fa fa-check"></i><b>4</b> A Brief Overview of Polygenic Adaptation</a>
<ul>
<li class="chapter" data-level="4.1" data-path="a-brief-overview-of-polygenic-adaptation.html"><a href="a-brief-overview-of-polygenic-adaptation.html#overview-3"><i class="fa fa-check"></i><b>4.1</b> Overview</a></li>
<li class="chapter" data-level="4.2" data-path="a-brief-overview-of-polygenic-adaptation.html"><a href="a-brief-overview-of-polygenic-adaptation.html#what-is-polygenic-adaptation"><i class="fa fa-check"></i><b>4.2</b> What is polygenic adaptation?</a></li>
<li class="chapter" data-level="4.3" data-path="a-brief-overview-of-polygenic-adaptation.html"><a href="a-brief-overview-of-polygenic-adaptation.html#fishers-geometrical-model-of-adaptation"><i class="fa fa-check"></i><b>4.3</b> Fisher’s Geometrical model of adaptation</a></li>
<li class="chapter" data-level="4.4" data-path="a-brief-overview-of-polygenic-adaptation.html"><a href="a-brief-overview-of-polygenic-adaptation.html#references-1"><i class="fa fa-check"></i><b>4.4</b> References</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="polygenic-adaptation-in-slim.html"><a href="polygenic-adaptation-in-slim.html"><i class="fa fa-check"></i><b>5</b> Polygenic Adaptation in SLiM</a>
<ul>
<li class="chapter" data-level="5.1" data-path="polygenic-adaptation-in-slim.html"><a href="polygenic-adaptation-in-slim.html#overview-4"><i class="fa fa-check"></i><b>5.1</b> Overview</a></li>
<li class="chapter" data-level="5.2" data-path="polygenic-adaptation-in-slim.html"><a href="polygenic-adaptation-in-slim.html#a-single-polygenic-trait-chp5-1_1t.slim"><i class="fa fa-check"></i><b>5.2</b> A Single Polygenic Trait (Chp5-1_1T.slim)</a></li>
<li class="chapter" data-level="5.3" data-path="polygenic-adaptation-in-slim.html"><a href="polygenic-adaptation-in-slim.html#parameters"><i class="fa fa-check"></i><b>5.3</b> Parameters</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="polygenic-adaptation-in-slim.html"><a href="polygenic-adaptation-in-slim.html#ne"><i class="fa fa-check"></i><b>5.3.1</b> Ne</a></li>
<li class="chapter" data-level="5.3.2" data-path="polygenic-adaptation-in-slim.html"><a href="polygenic-adaptation-in-slim.html#del_mean-and-del_shape"><i class="fa fa-check"></i><b>5.3.2</b> del_mean and del_shape</a></li>
<li class="chapter" data-level="5.3.3" data-path="polygenic-adaptation-in-slim.html"><a href="polygenic-adaptation-in-slim.html#mutweights"><i class="fa fa-check"></i><b>5.3.3</b> mutWeights</a></li>
<li class="chapter" data-level="5.3.4" data-path="polygenic-adaptation-in-slim.html"><a href="polygenic-adaptation-in-slim.html#rwide"><i class="fa fa-check"></i><b>5.3.4</b> rwide</a></li>
<li class="chapter" data-level="5.3.5" data-path="polygenic-adaptation-in-slim.html"><a href="polygenic-adaptation-in-slim.html#genomelength"><i class="fa fa-check"></i><b>5.3.5</b> genomelength</a></li>
<li class="chapter" data-level="5.3.6" data-path="polygenic-adaptation-in-slim.html"><a href="polygenic-adaptation-in-slim.html#locimu-locisigma-and-locidist"><i class="fa fa-check"></i><b>5.3.6</b> locimu, locisigma, and locidist</a></li>
<li class="chapter" data-level="5.3.7" data-path="polygenic-adaptation-in-slim.html"><a href="polygenic-adaptation-in-slim.html#width"><i class="fa fa-check"></i><b>5.3.7</b> width</a></li>
<li class="chapter" data-level="5.3.8" data-path="polygenic-adaptation-in-slim.html"><a href="polygenic-adaptation-in-slim.html#printh"><i class="fa fa-check"></i><b>5.3.8</b> printH</a></li>
<li class="chapter" data-level="5.3.9" data-path="polygenic-adaptation-in-slim.html"><a href="polygenic-adaptation-in-slim.html#samplerate"><i class="fa fa-check"></i><b>5.3.9</b> samplerate</a></li>
<li class="chapter" data-level="5.3.10" data-path="polygenic-adaptation-in-slim.html"><a href="polygenic-adaptation-in-slim.html#modelindex"><i class="fa fa-check"></i><b>5.3.10</b> modelindex</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="running-slim-in-parallel.html"><a href="running-slim-in-parallel.html"><i class="fa fa-check"></i><b>6</b> Running SLiM in Parallel</a>
<ul>
<li class="chapter" data-level="6.1" data-path="running-slim-in-parallel.html"><a href="running-slim-in-parallel.html#overview-5"><i class="fa fa-check"></i><b>6.1</b> Overview</a></li>
<li class="chapter" data-level="6.2" data-path="running-slim-in-parallel.html"><a href="running-slim-in-parallel.html#slim-at-the-command-line"><i class="fa fa-check"></i><b>6.2</b> SLiM at the Command Line</a></li>
<li class="chapter" data-level="6.3" data-path="running-slim-in-parallel.html"><a href="running-slim-in-parallel.html#running-slim-via-bash"><i class="fa fa-check"></i><b>6.3</b> Running SLiM via Bash</a></li>
<li class="chapter" data-level="6.4" data-path="running-slim-in-parallel.html"><a href="running-slim-in-parallel.html#running-slim-via-r"><i class="fa fa-check"></i><b>6.4</b> Running SLiM via R</a></li>
<li class="chapter" data-level="6.5" data-path="running-slim-in-parallel.html"><a href="running-slim-in-parallel.html#running-slim-in-python"><i class="fa fa-check"></i><b>6.5</b> Running SLiM in Python</a></li>
<li class="chapter" data-level="6.6" data-path="running-slim-in-parallel.html"><a href="running-slim-in-parallel.html#running-slim-via-a-c-program"><i class="fa fa-check"></i><b>6.6</b> Running SLiM via a C++ Program</a></li>
<li class="chapter" data-level="6.7" data-path="running-slim-in-parallel.html"><a href="running-slim-in-parallel.html#writing-slim-code-with-parallelism-in-mind"><i class="fa fa-check"></i><b>6.7</b> Writing SLiM code with parallelism in mind</a>
<ul>
<li class="chapter" data-level="6.7.1" data-path="running-slim-in-parallel.html"><a href="running-slim-in-parallel.html#single-file-output"><i class="fa fa-check"></i><b>6.7.1</b> Single file output</a></li>
<li class="chapter" data-level="6.7.2" data-path="running-slim-in-parallel.html"><a href="running-slim-in-parallel.html#multiple-file-output"><i class="fa fa-check"></i><b>6.7.2</b> Multiple file output</a></li>
</ul></li>
<li class="chapter" data-level="6.8" data-path="running-slim-in-parallel.html"><a href="running-slim-in-parallel.html#footnotes-1"><i class="fa fa-check"></i><b>6.8</b> Footnotes</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="running-slim-on-a-hpc-cluster.html"><a href="running-slim-on-a-hpc-cluster.html"><i class="fa fa-check"></i><b>7</b> Running SLiM on a HPC Cluster</a>
<ul>
<li class="chapter" data-level="7.1" data-path="running-slim-on-a-hpc-cluster.html"><a href="running-slim-on-a-hpc-cluster.html#overview-6"><i class="fa fa-check"></i><b>7.1</b> Overview</a></li>
<li class="chapter" data-level="7.2" data-path="running-slim-on-a-hpc-cluster.html"><a href="running-slim-on-a-hpc-cluster.html#connecting-to-tinaroo-and-set-up"><i class="fa fa-check"></i><b>7.2</b> Connecting to Tinaroo and Set-up</a></li>
<li class="chapter" data-level="7.3" data-path="running-slim-on-a-hpc-cluster.html"><a href="running-slim-on-a-hpc-cluster.html#pbs-scripts"><i class="fa fa-check"></i><b>7.3</b> PBS Scripts</a></li>
<li class="chapter" data-level="7.4" data-path="running-slim-on-a-hpc-cluster.html"><a href="running-slim-on-a-hpc-cluster.html#multi-node-jobs"><i class="fa fa-check"></i><b>7.4</b> Multi-node jobs</a>
<ul>
<li class="chapter" data-level="7.4.1" data-path="running-slim-on-a-hpc-cluster.html"><a href="running-slim-on-a-hpc-cluster.html#job-array-slim-jobs"><i class="fa fa-check"></i><b>7.4.1</b> Job array SLiM jobs</a></li>
<li class="chapter" data-level="7.4.2" data-path="running-slim-on-a-hpc-cluster.html"><a href="running-slim-on-a-hpc-cluster.html#embedded-nimrod-slim-jobs"><i class="fa fa-check"></i><b>7.4.2</b> Embedded Nimrod SLiM jobs</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="running-slim-on-a-hpc-cluster.html"><a href="running-slim-on-a-hpc-cluster.html#estimating-simulation-time"><i class="fa fa-check"></i><b>7.5</b> Estimating Simulation Time</a></li>
<li class="chapter" data-level="7.6" data-path="running-slim-on-a-hpc-cluster.html"><a href="running-slim-on-a-hpc-cluster.html#other-considerations"><i class="fa fa-check"></i><b>7.6</b> Other Considerations</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="latin-hypercube-sampling.html"><a href="latin-hypercube-sampling.html"><i class="fa fa-check"></i><b>8</b> Latin Hypercube Sampling</a>
<ul>
<li class="chapter" data-level="8.1" data-path="latin-hypercube-sampling.html"><a href="latin-hypercube-sampling.html#overview-7"><i class="fa fa-check"></i><b>8.1</b> Overview</a></li>
<li class="chapter" data-level="8.2" data-path="latin-hypercube-sampling.html"><a href="latin-hypercube-sampling.html#what-is-latin-hypercube-sampling"><i class="fa fa-check"></i><b>8.2</b> What is Latin hypercube sampling?</a></li>
<li class="chapter" data-level="8.3" data-path="latin-hypercube-sampling.html"><a href="latin-hypercube-sampling.html#generating-hypercubes-in-r"><i class="fa fa-check"></i><b>8.3</b> Generating hypercubes in R</a></li>
<li class="chapter" data-level="8.4" data-path="latin-hypercube-sampling.html"><a href="latin-hypercube-sampling.html#running-slim-with-hypercube-parameters"><i class="fa fa-check"></i><b>8.4</b> Running SLiM with hypercube parameters</a></li>
<li class="chapter" data-level="8.5" data-path="latin-hypercube-sampling.html"><a href="latin-hypercube-sampling.html#considerations"><i class="fa fa-check"></i><b>8.5</b> Considerations</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Polygenic SLiMulations: Investigating polygenic adaptation with SLiM 3.</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="polygenic-adaptation-in-slim" class="section level1" number="5">
<h1><span class="header-section-number">5</span> Polygenic Adaptation in SLiM</h1>
<div id="overview-4" class="section level2" number="5.1">
<h2><span class="header-section-number">5.1</span> Overview</h2>
<p>In this chapter I’ll provide a number of templates aiming to reconstruct some of the polygenic models we have seen in SLiM.
I’ve aimed to make these templates highly customisable, with many features that can be adjusted as you see fit.
As such, I’ll break down each section of the code to explain what it does, which may give you ideas for how it could be extended for your own use cases,
or which parts can be removed if you don’t need them and want to maximise your speed and/or minimise memory use.</p>
</div>
<div id="a-single-polygenic-trait-chp5-1_1t.slim" class="section level2" number="5.2">
<h2><span class="header-section-number">5.2</span> A Single Polygenic Trait (Chp5-1_1T.slim)</h2>
<p>The first example we’ll go over models a single polygenic trait evolving through either drift, stabilising selection, or directional selection.
This template has been built to be very broad, with a large array of parameters. These parameters are defined at the top of the file -
any line beginning with <code>SetCfgParam()</code> is a different parameter you may adjust<span class="math inline">\(^*\)</span>. The defaults are set based on realistic values, with sources
cited in comments. See Box 5.2.1 for information on each of these parameters.</p>
<p><span class="math inline">\(^*\)</span> : <code>SetCfgParam()</code> is based on the SLiM function <code>defineConstant()</code>, but works for both command line and SLiMgui so no changes are
necessary to switch between the two.</p>
<p>The model begins with a specified number of generations of burn-in (given via the parameter <code>burnTime</code>).
This burn-in period involves mutations subject to neutral drift, which yields standing genetic variation for selection to act on when a fitness function
is applied. The idea is for the population to be at mutation-drift equilibrium by the time the selection regime is applied.
I’ve found this typically occurs within 50,000 generations for a population size of 8,000, but you’ll likely want to adjust this as you see fit.
One way to confirm when a population is at equilibrium is by tracking heterozygosity over time. In the SLiM script (Chp5-1_1T.slim) you can
do this by setting <code>SetCfgParam(printH, T);</code> (Line 36). This will plot heterozygosity over time during your set burn-in period for you to view.
When the population is at equilibrium, this value should remain relatively stable around <span class="math inline">\(\theta = 4N_e\mu\)</span>, however the amount of stability around
this value (or any value at all) is dependent on several parameters including linkage (given by <code>rwide</code>), and background selection
(determined by <code>del_mean</code>, <code>del_shape</code>, and `<code>mutWeights</code>). During burn-in, a variety of output files are generated.
These include files containing
- Information on which genome positions are QTLs as opposed to non-QTL sites
- Heterozygosity and phenotypic means over time (sampled every X generations, defined by the first value in <code>samplerate</code>).
- The phenotypic location of the trait optimum.</p>
<p>Following burn-in, a certain selection regime takes place for a specified number of generations. This is set using the <code>selType</code> and
<code>testTime</code> parameters. Here, the population follows a stabilising or directinal selection trajectory, or continues via neutral drift
(“Brownian motion”). During this process, some more output is generated:
- Information on heterozygosity, phenotypic means, and fitness (sampled at a rate fiven by the first value in <code>samplerate</code>)
- Allelic effect sizes, frequencies, fixation times, etc. (sampled at a rate given by the second value in <code>samplerate</code>)</p>
<p>Note that mutation output is quite large (seeing as it stores many lines every time it writes to file), so you will want to adjust the sampling
frequency as per your available storage, the size of your simulation (in number of generations, size of the genome, number of mutations/mutation rate etc.).</p>
</div>
<div id="parameters" class="section level2" number="5.3">
<h2><span class="header-section-number">5.3</span> Parameters</h2>
<p>The script template contains a lot of parameters. Some of these will be applicable to you, others won’t be. I’ll explain each of them
here so you can get an idea. In brackets next to the parameter name is the type it expects, and how many arguments it expects (if &gt; 1).</p>
<table>
<colgroup>
<col width="30%" />
<col width="69%" />
</colgroup>
<thead>
<tr class="header">
<th>Parameter (type)</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>mu (float)</td>
<td>Genome-wide mutation rate</td>
</tr>
<tr class="even">
<td>Ne (int)</td>
<td>Population size</td>
</tr>
<tr class="odd">
<td>del_mean and del_shape (float)</td>
<td>Mean and shape parameters used to pull effects of deleterious mutations</td>
</tr>
<tr class="even">
<td>mutWeights (float, 3)</td>
<td>Proportions of mutations of different types</td>
</tr>
<tr class="odd">
<td>rwide (float)</td>
<td>Genome wide recombination rate</td>
</tr>
<tr class="even">
<td>nloci (int)</td>
<td>The number of QTLs to model</td>
</tr>
<tr class="odd">
<td>genomelength (int)</td>
<td>The total length of the genome in loci</td>
</tr>
<tr class="even">
<td>locimu, locisigma, locidist (float, float, string)</td>
<td>Parameters for QTL distribution of additive effects.</td>
</tr>
<tr class="odd">
<td>width (float)</td>
<td>Stabilising selection fitness function width parameter</td>
</tr>
<tr class="even">
<td>opt (numeric)</td>
<td>Position of the optimum relative to the burn-in phenotype mean</td>
</tr>
<tr class="odd">
<td>selType (string)</td>
<td>The type of selection to use during the test: “s” = stabilising, “p” = positive/directional, “d” = drift</td>
</tr>
<tr class="even">
<td>printH (bool)</td>
<td>Draw a graph of heterozygosity during burn-in to diagnose mutation-drift equilibrium</td>
</tr>
<tr class="odd">
<td>burnTime (int)</td>
<td>Number of generations to run neutral burn-in for.</td>
</tr>
<tr class="even">
<td>testTime (int)</td>
<td>Number of generations to run the selective regime for.</td>
</tr>
<tr class="odd">
<td>samplerate (int, 2)</td>
<td>How often, in generations, to save output</td>
</tr>
<tr class="even">
<td>out… (string)</td>
<td>Output filenames and paths for various output files</td>
</tr>
<tr class="odd">
<td>modelindex (int)</td>
<td>Identifier for Latin Hypercube job arrays</td>
</tr>
</tbody>
</table>
<p>Below are some tips regarding some of the parameters to guide you in designing your experiments:</p>
<div id="ne" class="section level3" number="5.3.1">
<h3><span class="header-section-number">5.3.1</span> Ne</h3>
<p>The single most expensive thing to do in SLiM is increase the population size. Be careful with setting this and time your
simulations as you adjust it so you can maximise your population size for the available time you have to run your simulations.
Be aware that smaller population sizes = stronger drift, and so selection will be comparatively weaker in small populations.</p>
</div>
<div id="del_mean-and-del_shape" class="section level3" number="5.3.2">
<h3><span class="header-section-number">5.3.2</span> del_mean and del_shape</h3>
<p>When mutations appear in SLiM, a fitness effect is pulled from a user-specified distribution of fitness effects.
Individuals with the mutation have their fitness adjusted according to this effect. Non-trait deleterious mutations are pulled
from a gamma distribution in this simulation, which asks for a mean and shape parameter. This is slightly different to the
standard mathematical notation, which usually specifies a <span class="math inline">\(\alpha\)</span> and <span class="math inline">\(\beta\)</span> parameter. In this case, the mean is equal to
<span class="math inline">\(\alpha\beta\)</span>, and the shape parameter is <span class="math inline">\(\alpha\)</span>. You are able to see the shape of your distribution of effects for any mutation
type by clicking the hand underneath the generation counter in SLiMgui and hovering over the mutation type:</p>
<p><img src="images/chapter5/box521_dfe1.png" /> <img src="images/chapter5/box521_dfe2.png" /></p>
</div>
<div id="mutweights" class="section level3" number="5.3.3">
<h3><span class="header-section-number">5.3.3</span> mutWeights</h3>
<p>In this simulation, not all genome positions are QTLs. Some are sites with nothing to do with the trait whatsoever.
Therefore, there are 3 mutation types: neutral (m1), which do nothing except give an idea of how linked variation
declines with other types; non-trait deleterious (m2), which act as a form of background selection: imagine these as
being mutations that don’t affect the trait but reduce fitness; and QTLs (m3), which affect the trait. All three of
these types can occur at a QTL <em>site</em>, but QTL <em>mutations</em> (m3) cannot occur at non-QTL <em>sites</em>. Imagine a mutation
occurs at a QTL that causes a pleiotropic effect elsewhere, resulting in an extremely deleterious effect. This is the
kind of situation this is modelling. Similarly, a neutral mutation could occur at a QTL <em>site</em> that is synonymous,
not changing the phenotype at all.
In this simulation, all mutations begin life as an m1 mutation. The simulation checks if any mutation exists at a QTL site
and then randomly chooses a type based on <code>mutWeights</code>. This parameter specifies the relative likelihood of a
given mutation type (neutral vs QTL vs deleterious) occurring.</p>
</div>
<div id="rwide" class="section level3" number="5.3.4">
<h3><span class="header-section-number">5.3.4</span> rwide</h3>
<p>From my experience, recombination rate in SLiM doesn’t typically affect performance too much when using realistic values.
However, if you want to model independent assortment (by setting <code>rwide</code> to 0.5) to get rid of linkage effects,
you will probably see a tiny<span class="math inline">\(^1\)</span> performance drop<span class="math inline">\(^2\)</span>. I’d suggest testing this out, and if you want to model independent
assortment, do so with a small genome (or a small population). This parameter is set as a single genome-wide value, however
recombination maps can be constructed in SLiM using a vector of rates followed by a vector of endpoints where the next rate
begins. This will require some small adjustments to the script, but should be trivial given you read the documentation on the
<code>initializeRecombinationRate()</code> method.</p>
<p><font size = "1"><span class="math inline">\(^1\)</span>: Massive</p>
<p><span class="math inline">\(^2\)</span>: Death of your computer</font></p>
</div>
<div id="genomelength" class="section level3" number="5.3.5">
<h3><span class="header-section-number">5.3.5</span> genomelength</h3>
<p>This simulation assumes each ‘box’ in the SLiM genome is a locus rather than a base pair. This means all rates (rwide, mu etc.)
are scaled according to per locus per generation, rather than per base pair. <code>nloci</code> specifies the number of QTL positions
to be randomly scattered across the genome, whereas genomelength is the actual genome length in total loci.
Hence, <code>nloci</code> must be <span class="math inline">\(&lt;=\)</span> <code>genomelength</code>.</p>
</div>
<div id="locimu-locisigma-and-locidist" class="section level3" number="5.3.6">
<h3><span class="header-section-number">5.3.6</span> locimu, locisigma, and locidist</h3>
<p>QTL mutations effects are pulled from a normal distribution by default in this simulation, with mean = locimu and variance = locisigma.
This can be changed though, using the <code>locidist</code> parameter. A normal distribution (default) is set using “n”.
A gamma distribution of effects can also be chosen using “g”.</p>
</div>
<div id="width" class="section level3" number="5.3.7">
<h3><span class="header-section-number">5.3.7</span> width</h3>
<p>The fitness function being used for stabilising selection comes from Lande 1976 (SOURCE).
This equation comes out to the form <span class="math inline">\(f(z)\ =\ 0.01+\exp(-(z^{2})*\left(\frac{1}{2w^{2}}\right))\)</span></p>
<p>Where f(z) is the fitness of an individual z units from the optimum, w is the width of the function, and z is the Euclidean distance from the optimum.
You can play around with that <a href="https://www.desmos.com/calculator/yelcak6wfo">here</a>. Moving the width will show
you how fitness decays as an individual gets far from the optimum: higher width means weaker selection
(lower difference between individuals at the optimum vs those far away).</p>
<p>The <code>width</code> parameter is precomputed to save a little bit of computing time - computers don’t like doing division
very much, and doing that every time we calculate fitness is wasteful.
Hence, we precompute <span class="math inline">\(\frac{1}{2w^{2}}\)</span>, and that is our width parameter.</p>
</div>
<div id="printh" class="section level3" number="5.3.8">
<h3><span class="header-section-number">5.3.8</span> printH</h3>
<p>Setting this option to <code>T</code> will enable a diagnostic plot of heterozygosity over time to help visualise mutation-drift
equilibrium for your given mutation rate and population size (see Section 5.2). Dotted lines drawn indicate 10% CIs of <span class="math inline">\(\theta\)</span>,
a measure of nucleotide diversity given by <span class="math inline">\(\theta = 4N_e\mu\)</span>.</p>
</div>
<div id="samplerate" class="section level3" number="5.3.9">
<h3><span class="header-section-number">5.3.9</span> samplerate</h3>
<p>There are two values set in the <code>samplerate</code> parameter. The first value corresponds to ordinary outputs
(means, heterozygosities etc.), which occur during burn-in and the test. The second value corresponds to mutation output,
which is considerably large due to printing many lines per generation. As a result, you likely will need to sample mutations less
frequently than other output in order to save space (and have your files openable in R without running out of memory).</p>
</div>
<div id="modelindex" class="section level3" number="5.3.10">
<h3><span class="header-section-number">5.3.10</span> modelindex</h3>
<p><code>modelindex</code> is an important parameter for Latin Hypercube sets of simulations, however we’ll get into that later.
For now, you can use it individually to quickly identify different simulations where you have changed a bunch of
parameters (e.g. if you have three levels of <code>rwide</code>, you could set three different <code>modelindex</code> values to
identify those models. This is more useful when combined with interactions between multiple parameters). It can also be
used as a proxy for all of your variables, saving considerable space when storing output. For example, you will only need
one column to identify the parameters which led to a result rather than a column for each parameter. This is particularly
input with regards to mutation output, which grows to potentially 100s of gigabytes in size.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="a-brief-overview-of-polygenic-adaptation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="running-slim-in-parallel.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
